src/ReflectionClass.php

```diff
@@ -112,9 +134,7 @@ public static function collectTraitsFromClassNode(ClassLike $classLikeNode, arra
      */
     public function __debugInfo(): array
     {
-        return [
-            'name' => $this->getName()
-        ];
+        return [];
```

Comment:
What exactly is `__debugInfo()` for?

Reply:
This method is called by `var_dump()` and similar tools (including IDE too) to return an array with debug data of object.
Mainly it is used to hide complex internal data structure (eg low-level AST tree nodes, variables, etc) and return fake data to make object looks like it has one simple property, in this case - name. Better for logs, but worse for debugging )

See an example here: https://3v4l.org/FF7RY

Reply:
Ohhh I see, thanks for the information!

Reply:
Re-added

https://github.com/goaop/parser-reflection/pull/122/commits/1585b64af3ce5d2ec0595d1d2a378fedc51ec635

Reply:
Added description
https://github.com/goaop/parser-reflection/pull/122/commits/f8dd6098edbe29776fcbe5fa0d032f58b1b31a95

Reply:
When debugging the ReflectionClass constructor, sometimes the debugger crashes, because the ide or the debugger calls this method internally and the `getName` method in `__debugInfo` should always return a string, but throws an Exception because the constructor wasn't done yet. What should the `var_dump()` method return while inside the constructor? Just `null` values or and `Uninitialized` string?

Reply:
Yep, this can happen, indeed. 
Inside ctor, you can just return empty array or just return null for value of name (which is better and logical) to prevent exception throwing.